crons_monitor_data.go

```diff
@@ -57,6 +59,41 @@ func NewCronsMonitorData(monitorSlug string, schedule string, maxRunTime int64,
 
 // Add a job to the crons monitor
 func (c *CronsMonitorData) addJob(job *batchv1.Job, checkinId sentry.EventID) error {
+	c.mutex.Lock()
```

Comment:
A common pattern is to do
```
c.mutex.Lock()
defer c.mutex.Unlock()
```

...to visually localize lock operations, and make sure that the mutex will be unlocked when we leave the function (even if there's an error).